aten/src/ATen/native/cuda/ReduceSumProdKernel.cu

```diff
@@ -41,6 +48,26 @@ static void sum_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+static void nansum_kernel_cuda(TensorIterator& iter) {
```

Comment:
This code is idential to the sum_kernel_cuda code except for the use of nansum_kernel_impl, right?

Reply:
Yes, but I am not sure if the common structure can be factored out and pass `nansum_kernel_impl` as it is an uninitialized template at that point.

Note: prod also have same structure, except for `prod_kernel_impl`.

Reply:
Another difference is that they support different types based on the dispatch
```
sum -> AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND
nansum -> AT_DISPATCH_FLOATING_TYPES // c10::complex didn't have an overload for ::isnan last I checked.
prod -> AT_DISPATCH_ALL_TYPES
```

Reply:
Have refactored it. Please review :)